apps/HowtoAddNewApp.md

```diff
@@ -0,0 +1,192 @@
+# How to add new application
+
+The following files are required to integrate a new application with the Tizen RT build system:  
+- **Source file**  
+- [**Kconfig**](#kconfig)  
+- [**Kconfig_ENTRY**](#kconfig_entry)  
+- [**Make.defs**](#makedefs)  
+- [**Makefile**](#makefile)
+
+## Kconfig
+This file is required by Kconfig-frontends menu build system.  
+It is used to specify various config values according to which the application should be built.  
+The following two config values must be defined in the Kconfig file.
+```
+config <APP_NAME>
+	bool "<Write Description>"
+	default n
+	---help---
+		<Write Details>
+
+config USER_ENTRYPOINT
+	string
+	default "<APP_MAIN_FUNCTION>" if ENTRY_<APP_NAME>
+```
+The first config parameter specifies whether the application should be built or not.
+
+The second config parameter specifies the function name of the application which should be executed when OS executes the application. The entrypoint can be set manually but for it the name of function which has to be executed firstly should be known for each application. This option registers the app with the build system and can be selected from the menu to be used as application entrypoint.
+
+## Kconfig_ENTRY
+This contains only one configuration which allows entry point selection combined above USER_ENTRYPOINT configuration.
+```
+config ENTRY_<APP_NAME>
+	bool "<Write Description>"
+	depends on <APP_NAME>
+```
+
+Menuconfig seqeunce is shown below:
```

Comment:
Steps in menuconfig to enable and selecting an app as userentry point  are shown below